### PR TITLE
Pass missing and default in deserialized form

### LINKED ISSUE #378: missing and default values should be specified in deserialized form
Marshmallow expects `missing` and `default` values to be specified in a pre-serialized form, which is inconvenient and unintuitive.

Here is a schema which I would expect to work:

``` python
import datetime
import uuid

class TestSchema(Schema):
    id = fields.UUID(
        missing=uuid.uuid1)
    ts = fields.DateTime(
        format='rfc',
        missing=datetime.datetime.now)
```

However, it does not:

``` python
schema = TestSchema()
pprint.pprint(tuple(schema.load({})))
({}, {'id': [u'Not a valid UUID.'], 'ts': [u'Not a valid datetime.']})
```

Instead, I have to convert my `default` and `missing` values to primitive types, which undermines the purpose of a library like marshmallow that already knows how to do this.  For data types like datetimes, which have many possible serialized formats, I have to keep the formatting of my `default` and `missing` values in sync with my field properties, which is fragile and convoluted:

``` python
from email.Utils import formatdate

class TestSchema(Schema):
    id = fields.UUID(
        missing=lambda: str(uuid.uuid1()))
    ts = fields.DateTime(
        format='rfc',
        missing=formatdate)
```

In the case of `default` values, marshmallow does not attempt to validate them, so no error is raised, but the resulting document is not deserializable.

### PR DESCRIPTION
**Pull request by lafrech**
Closes #378

### CODE REVIEW

**Comment by lafrech** [comment]
TODO: update docs.

**Comment by sloria** [comment]
Behavior looks good. This should be good to merge after docs are updated.

**Comment by lafrech** [comment]
Rebased, added a note in "upgrading" and presented the feature in "quickstart".

File: docs/quickstart.rst [hunk 1]
```diff
@@ -481,6 +481,25 @@ In the context of a web API, the ``dump_only`` and ``load_only`` parameters are
         created_at = fields.DateTime(dump_only=True)
 
 
+Specify default Serialization/Deserialization values
```
Comment (sloria) [inline docs/quickstart.rst:484]: "Default" here should be capitalized. I'll fix this.

Comment (lafrech) [reply docs/quickstart.rst:484]: Oh. I'm not sure what the rule is. Capitalize everything except common "little" words like "and", "as" "the", "a",...?

Comment (sloria) [reply docs/quickstart.rst:484]: Correct. Headings are title cased.

**Comment by taion** [comment]
This creates an interesting breaking change in behavior when both `required` and `missing` are specified. Before this change, `required` would essentially have no effect if `missing` was specified, as the default would get applied before field-level deserialization.

After this change, `missing` now has no effect if `required` is specified, as the `required` check is made before the `missing` logic is applied.

I think it would make more sense to restore the old behavior – as with `missing` implicitly overriding `allow_none`, it should also override `required`.

**Pull request merged**